macOS/DuckDuckGo/NavigationBar/View/AddressBarTextField.swift

```diff
@@ -1280,3 +1240,51 @@ extension AddressBarTextField: SuggestionViewControllerDelegate {
 fileprivate extension NSStoryboard {
     static let suggestion = NSStoryboard(name: "Suggestion", bundle: .main)
 }
+
+extension URL {
+
+    static func makeUrl(suggestion: Suggestion?, stringValueWithoutSuffix: String, completion: @escaping (URL?, String, Bool) -> Void) {
+        let finalUrl: URL?
+        let userEnteredValue: String
+        switch suggestion {
+        case .bookmark(title: _, url: let url, isFavorite: _, _),
+             .historyEntry(title: _, url: let url, _),
+             .website(url: let url),
+             .internalPage(title: _, url: let url, _),
+             .openTab(title: _, url: let url, _, _):
+            finalUrl = url
+            userEnteredValue = url.absoluteString
+        case .phrase(phrase: let phrase),
+             .unknown(value: let phrase):
+            finalUrl = URL.makeSearchUrl(from: phrase)
+            userEnteredValue = phrase
+        case .none:
+            finalUrl = URL.makeURL(from: stringValueWithoutSuffix)
+            userEnteredValue = stringValueWithoutSuffix
+        }
+
+        guard let url = finalUrl else {
+            completion(finalUrl, userEnteredValue, false)
+            return
+        }
+
+        upgradeToHttps(url: url, userEnteredValue: userEnteredValue, completion: completion)
+    }
+
+    static private func upgradeToHttps(url: URL, userEnteredValue: String, completion: @escaping (URL?, String, Bool) -> Void) {
+        Task {
+            let result = await NSApp.delegateTyped.privacyFeatures.httpsUpgrade.upgrade(url: url)
+            switch result {
+            case let .success(upgradedUrl):
+                DispatchQueue.main.async {
```

Comment:
There was also a bug where the completion handler was being called from a background thread after upgrading the URL.